src/decode.rs

```diff
@@ -4685,16 +4687,18 @@ unsafe fn decode_b(
                         f.prev_segmap,
                         f.b4_stride,
                     );
-                    if seg_id_0 >= 8 as libc::c_int as libc::c_uint {
-                        return -(1 as libc::c_int);
+
+                    if seg_id >= 8 {
+                        return -1;
                     }
```

Comment:
Ah, yes, it is bounded, so https://github.com/memorysafety/rav1d/pull/145/files#r1169357964 is fine then.